homeassistant/components/smarttub/entity.py

```diff
@@ -54,3 +54,24 @@ def __init__(self, coordinator, spa, sensor_name, state_key):
     def _state(self):
         """Retrieve the underlying state from the spa."""
         return getattr(self.spa_status, self._state_key)
+
+
+class SmartTubExternalSensorBase(SmartTubEntity):
+    """Class for additional BLE wireless sensors sold separately."""
+
+    def __init__(self, coordinator, spa, sensor: smarttub.SpaSensor) -> None:
+        """Initialize the external sensor entity."""
+        self.sensor_address = sensor.address
+        self._attr_unique_id = f"{spa.id}-externalsensor-{sensor.address}"
+        super().__init__(coordinator, spa, self._human_readable_name(sensor))
+
+    @staticmethod
+    def _human_readable_name(sensor):
```

Comment:
So, wouldn't this also warrant a separate device? We can connect it to the spa using `via_device` but I think it makes more sense to use here